ide/nativeimage.api/src/org/netbeans/modules/nativeimage/spi/debug/NIDebuggerProvider.java

```diff
@@ -120,4 +124,24 @@
      * @since 1.0
      */
     String getVersion();
+
+    /**
+     * Provide a list of locations for a given file path.
+     *
+     * @param filePath a file path
+     * @return list of locations, or <code>null</code> when there's no location
+     *         information about such file.
+     * @since 0.2
+     */
+    default List<Location> listLocations(String filePath) {
```

Comment:
Consider adding subinterface `SymbolProvider extends Provider`

Reply:
There's just one implementation and one module that uses it. I'd rather keep the single provider till it grows more...